src/scenic/core/regions.py

```diff
@@ -281,883 +306,2816 @@ def __hash__(self):
+        width (float): width of the rectangle.
+        length (float): length of the rectangle.
+        name (str; optional): name for debugging.
+    """
+    def __init__(self, position, heading, width, length, name=None):
+        self.position = toVector(position, "position of RectangularRegion not a vector")
+        self.heading = toScalar(heading, "heading of RectangularRegion not a scalar")
+        self.width = toScalar(width, "width of RectangularRegion not a scalar")
+        self.length = toScalar(length, "length of RectangularRegion not a scalar")
+
+        deps = [self.position, self.heading, self.width, self.length]
+
+        self.hw = hw = width / 2
+        self.hl = hl = length / 2
+        self.radius = hypot(hw, hl)     # circumcircle; for collision detection
+        self.corners = tuple(self.position.offsetRotated(heading, Vector(*offset))
+            for offset in ((hw, hl), (-hw, hl), (-hw, -hl), (hw, -hl)))
+        self.circumcircle = (self.position, self.radius)
+
+        if any(needsSampling(dep) or needsLazyEvaluation(dep) for dep in deps):
+            # Center and radius aren't fixed, so we'll wait to pass a polygon
+            super().__init__(polygon=None, name=name, additionalDeps=deps)
+            return
+
+        polygons = self._make_rectangle_polygons()
+        super().__init__(polygon=polygons, z=self.position.z, name=name, additionalDeps=deps)
+
+    def _make_rectangle_polygons(self):
+        position, heading, hw, hl = self.position, self.heading, self.hw, self.hl
+        assert not any(isLazy(c) for c in (position, heading, hw, hl))
+        corners = _RotatedRectangle.makeCorners(position.x, position.y, heading, hw, hl)
+        polygon = shapely.geometry.Polygon(corners)
+        return MultiPolygon([polygon])
+
+
+    def sampleGiven(self, value):
+        return RectangularRegion(value[self.position], value[self.heading],
+            value[self.width], value[self.length],
+            name=self.name)
+
+    def evaluateInner(self, context):
+        position = valueInContext(self.position, context)
+        heading = valueInContext(self.heading, context)
+        width = valueInContext(self.width, context)
+        length = valueInContext(self.length, context)
+        return RectangularRegion(position, heading, width, length, name=self.name)
+
+    def uniformPointInner(self):
+        hw, hl = self.hw, self.hl
+        rx = random.uniform(-hw, hw)
+        ry = random.uniform(-hl, hl)
+        pt = self.position.offsetRotated(self.heading, Vector(rx, ry, self.position.z))
+        return self.orient(pt)
+
+    @property
+    def AABB(self):
+        x, y, z = zip(*self.corners)
+        minx, maxx = findMinMax(x)
+        miny, maxy = findMinMax(y)
+        return ((minx, miny), (maxx, maxy), (self.z, self.z))
+
+    def __repr__(self):
+        return f'RectangularRegion({self.position!r},{self.heading!r},{self.width!r},{self.length!r})'
```

Comment:
What was the point of merging these two lines and breaking the line length limit?

Reply:
Looks like a change you made a couple months ago that never made it into Scenic-devel.